File: java/1078.occurrences-after-bigram.java
import java.util.ArrayList;
import java.util.List;

/*
 * @lc app=leetcode id=1078 lang=java
 *
 * [1078] Occurrences After Bigram
 */

// @lc code=start
class Solution {
    public String[] findOcurrences(String text, String first, String second) {
        String[] split_text = text.split(" ");
        List<String> res = new ArrayList<>();
        for (int i = 2, len = split_text.length; i < len; ++i) {
            if (split_text[i - 2].equals(first) && split_text[i - 1].equals(second)) {
                res.add(split_text[i]);
            }
        }
        return res.toArray(new String[0]);
    }
}
// @lc code=end

File: 1078.occurrences-after-bigram.cpp
/*
 * @lc app=leetcode id=1078 lang=cpp
 *
 * [1078] Occurrences After Bigram
 */
#include <string>
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<string> findOcurrences(string text, string first, string second)
    {
        text.push_back(' ');
        bool find_first = false;
        bool match = false;
        vector<string> res;
        int pos = 0;
        int len = text.size();
        while (pos != len)
        {
            auto next_pos = text.find(' ', pos);
            if (match)
            {
                res.push_back(text.substr(pos, next_pos - pos));
                match=false;
            }
            if (!find_first)
            {
                if (text.substr(pos, next_pos - pos) == first)
                {
                    find_first = true;
                }
                pos = next_pos + 1;
            }
            else
            {
                if (text.substr(pos, next_pos - pos) == second)
                {
                    match = true;
                    if (second != first)
                    {
                        find_first = false;
                    }
                }
                else if (text.substr(pos, next_pos - pos) != first)
                {
                    find_first = false;
                }
                pos = next_pos + 1;
            }
        }
        return res;
    }
};
// @lc code=end
